src/components/ec-radio-btn/ec-radio-btn.vue

```diff
@@ -0,0 +1,298 @@
+            'ec-radio-btn__description--is-single-line': isTextInline,
+            'ec-radio-btn__description--disabled': isDisabled,
+          }"
+          :title="isTextInline ? description : undefined"
+          data-test="ec-radio-btn__description"
+        >
+          <slot name="description">{{ description }}</slot>
+        </p>
+      </div>
+    </div>
+
+    <div
+      :id="errorId"
+      v-if="hasErrorMessage"
+      class="ec-radio-btn__error-text"
+      data-test="ec-radio-btn__error-text"
+    >
+      <slot name="error-message">{{ errorMessage }}</slot>
+    </div>
+  </div>
+
+</template>
+
+<script setup lang="ts">
+import {
+  computed, ref, useSlots,
+} from 'vue';
+
+import { getUid } from '../../utils/uid';
+import EcIcon from '../ec-icon';
+import type { RadioButtonEvents } from './types';
+import { RadioButtonEvent } from './types';
+
+interface RadioButtonProps {
+  value: string,
+  modelValue?: string,
+  label?: string,
+  description?: string,
+  isDisabled?: boolean,
+  isTextInline?: boolean,
+  name?: string,
+  errorMessage?: string,
+  hasError?: boolean
+}
+
+const props = defineProps<RadioButtonProps>();
+
+const uid = getUid();
+const slots = useSlots();
+
+const id = `ec-radio-btn-${uid}`;
+const errorId = `error-${id}`;
+const labelId = `label-${id}`;
+const descriptionId = `description-${id}`;
+const inputRadioRef = ref();
+
+const emit = defineEmits<{(e: 'update:modelValue', value: RadioButtonEvents[RadioButtonEvent.UPDATE_MODEL_VALUE]): void,
+}>();
+
+const isChecked = computed(() => props.modelValue === props.value);
+const isFocused = ref(false);
+
+const hasLabel = computed(() => (!!props.label || !!slots.label));
+const hasDescription = computed(() => (!!props.description || !!slots.description));
+const hasErrorMessage = computed(() => (!!props.errorMessage || !!slots['error-message']));
+
+function setFocusState(state: boolean) {
+  isFocused.value = state;
+}
+
+function onClick() {
+  if (!props.isDisabled) {
+    inputRadioRef.value.focus();
+    emit(RadioButtonEvent.UPDATE_MODEL_VALUE, props.value);
+  }
+}
+</script>
+
+<style>
+.ec-radio-btn {
+  &__input {
+    @apply tw-sr-only;
+  }
+
+  &__wrapper {
+    @apply tw-flex tw-flex-nowrap;
+    @apply tw-content-center;
+
+    &--is-single-line {
+      @apply tw-items-center;
+    }
+  }
+
+  &__label {
+    @apply tw-cursor-pointer;
+    @apply tw-input-label;
+    @apply tw-flex-grow;
+    @apply tw-min-w-0;
+
+    &--disabled {
+      @apply tw-cursor-default;
+    }
+  }
+
+  &__description {
+    @apply tw-cursor-pointer;
+    @apply tw-small-text;
+    @apply tw-mb-0;
+    @apply tw-mt-8;
+
+    &--is-single-line {
+      @apply tw-mt-0;
+      @apply tw-ml-8;
+    }
+
+    &--disabled {
+      @apply tw-cursor-default;
+    }
+  }
+
+  &__text-wrapper {
+    @apply tw-cursor-pointer;
+
+    &--is-single-line {
+      @apply tw-flex;
+      @apply tw-flex-nowrap;
+    }
+  }
+
+  &__icon-wrapper {
+    @apply tw-cursor-pointer;
```

Comment:
we can set it to the root component element and remove from all other places

Reply:
this way the cursor pointer on the label is not working

Reply:
Let's add only for it another tw-cursor-pointer

Reply:
it doesn't really make sense